### PR TITLE
[FR-921] add support in api tokens

### PR DESCRIPTION
**Pull request by eldad-frontegg**

### CODE REVIEW

File: src/identity/with-authentication.ts [hunk 1]
```diff
@@ -54,7 +71,17 @@ export function withAuthentication({ roles = [], permissions = [] }: IWithAuthen
 
       // Store the decoded user on the request
       req.user = user;
-      req.user.id = user.sub; // The subject of the token (OpenID token) is saved on the req.user as well for easier readability
+      req.user.id = ''
+
+      switch (req.user.type) {
+        case 'userToken':
```
Comment (aviadmizrachi) [inline src/identity/with-authentication.ts:77]: Let's create consts

File: src/identity/contextResolver.ts [hunk 2]
```diff
@@ -3,15 +3,15 @@ import { FronteggPermissions } from '../../index';
 export const contextResolver = (req) => {
   if (!req.user) {
     return {
-      tenantId: '',
+      tenantId: 'NOT_A_REAL_TENANT_ID',
```
Comment (aviadmizrachi) [inline src/identity/contextResolver.ts:6]: WITHOUT_TENANT_ID

**Review by aviadmizrachi — CHANGES_REQUESTED** [review]

File: src/identity/contextResolver.ts (hunk 2 above)
Comment (eldad-frontegg) [reply src/identity/contextResolver.ts:6]: do you want me to change the name to "WITHOUT_TENANT_ID"? @aviadmizrachi

File: src/identity/with-authentication.ts [hunk 3]
```diff
@@ -54,7 +77,17 @@ export function withAuthentication({ roles = [], permissions = [] }: IWithAuthen
...
       // Store the decoded user on the request
       req.user = user;
-      req.user.id = user.sub; // The subject of the token (OpenID token) is saved on the req.user as well for easier readability
+      req.user.id = ''
+
+      switch (req.user.type) {
+        case tokenTypes.UserToken:
+          req.user.id = user.sub; // The subject of the token (OpenID token) is saved on the req.user as well for easier readability
+          break;
+        case tokenTypes.UserApiToken:
```
Comment (aviadmizrachi) [inline src/identity/with-authentication.ts:86]: We should add for the tenant api token as well.
Not a blocker but let's remember it

**Review by aviadmizrachi — APPROVED** [review]

**Pull request merged**